src/lib/libast/include/ast_common.h

```diff
@@ -0,0 +1,146 @@
+/* : : generated from common by iffe version 2013-11-14 : : */
+#ifndef _AST_COMMON_H
+#define _AST_COMMON_H	1
+#define _sys_types	1	/* #include <sys/types.h> ok */
+
+#define _std_proto	1	/* standard C prototypes ok */
+
+/* disable non-standard linux/gnu inlines */
+#ifdef __GNUC__
+#	undef	__OPTIMIZE_SIZE__
+#	define	__OPTIMIZE_SIZE__	1
+#endif
+
+/* __STD_C indicates that the language is ANSI-C or C++ */
+#if !defined(__STD_C) && __STDC__
+#	define	__STD_C		1
+#endif
+#if !defined(__STD_C) && (__cplusplus || c_plusplus)
+#	define __STD_C		1
+#endif
+#if !defined(__STD_C) && _std_proto
+#	define __STD_C		1
+#endif
+#if !defined(__STD_C)
+#	define __STD_C		0
+#endif
+
+/* extern symbols must be protected against C++ name mangling */
+#ifndef _BEGIN_EXTERNS_
+#	if __cplusplus || c_plusplus
+#		define _BEGIN_EXTERNS_	extern "C" {
+#		define _END_EXTERNS_	}
+#	else
+#		define _BEGIN_EXTERNS_
+#		define _END_EXTERNS_
+#	endif
+#endif
+
+/* _ARG_ simplifies function prototyping among flavors of C */
+#ifndef _ARG_
+#	if __STD_C
+#		define _ARG_(x)	x
+#	else
+#		define _ARG_(x)	()
+#	endif
+#endif
+
+/* __INLINE__, if defined, is the inline keyword */
+#if !defined(__INLINE__) && defined(__cplusplus)
+#	define __INLINE__	inline
+#endif
+
+/* Void_t is defined so that Void_t* can address any type */
+#ifndef Void_t
+#	if __STD_C
+#		define Void_t		void
+#	else
+#		define Void_t		char
+#	endif
+#endif
+
+/* windows variants and veneers */
+#if !defined(_WINIX) && (_UWIN || __CYGWIN__ || __EMX__)
```

Comment:
This should be shortened to
```c
#if !defined(_WINIX) && (__CYGWIN__)
```
Reason: Support for UWIN and __EMX__ (.i.e OS/2) is removed now.